src/payment/strategies/googlepay/googlepay.ts

```diff
@@ -0,0 +1,185 @@
+import { PaymentMethod } from '../..';
+import { Checkout } from '../../../checkout';
+import { BraintreeModule, BraintreeModuleCreator } from '../braintree';
+
+export type EnvironmentType = 'PRODUCTION' | 'TEST';
+type AddressFormat = 'FULL' | 'MIN';
+type TotalPriceStatus = 'ESTIMATED' | 'FINAL' | 'NOT_CURRENTLY_KNOWN';
+type TokenizeType = 'AndroidPayCard' | 'CreditCard';
+
+export interface GooglePayBraintreeSDK extends BraintreeModule {
+    createPaymentDataRequest(request?: GooglePayDataRequestV1): GooglePayPaymentDataRequestV1;
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload>;
+}
+
+export interface GooglePayInitializer {
+    initialize(checkout: Checkout, paymentMethod: PaymentMethod, hasShippingAddress: boolean, publishableKey?: string): Promise<GooglePayPaymentDataRequestV1>;
+    teardown(): Promise<void>;
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload>;
+}
+
+export interface GooglePayCreator extends BraintreeModuleCreator<GooglePayBraintreeSDK> {}
+
+export interface GooglePayPaymentOptions {
+    environment: EnvironmentType;
+}
+
+export interface GooglePayDataRequestV1 {
+    merchantInfo: {
+        authJwt?: string,
+    };
+    transactionInfo: {
+        currencyCode: string,
+        totalPriceStatus: TotalPriceStatus,
+        totalPrice: string,
+    };
+    cardRequirements: {
+        billingAddressRequired: boolean,
+        billingAddressFormat: AddressFormat,
+    };
+    emailRequired: boolean;
+    phoneNumberRequired: boolean;
+    shippingAddressRequired: boolean;
+}
+
+export interface GooglePayPaymentDataRequestV1 {
+    allowedPaymentMethods: string[];
+    apiVersion: number;
```

Comment:
We need to make sure that we use `apiVersion: 2` so that we support JWT platform solution.

Reply:
This will be work on https://jira.bigcommerce.com/browse/INT-982